### PR TITLE
Named DiagnosticSettings

### PR DESCRIPTION
**Pull request by vcanaa**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by sergey-shandar** [comment]
@ravbhatnagar please, have a look. It's a new API version.

File: specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/diagnosticsSettingsCategories_API.json
Comment (ravbhatnagar) [inline specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/diagnosticsSettingsCategories_API.json]: Please indicate some of the typical values resourceUri could take to help customers understand this.

File: specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/diagnosticsSettingsCategories_API.json
Comment (ravbhatnagar) [inline specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/diagnosticsSettingsCategories_API.json]: Make this an enum

File: specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/diagnosticsSettingsCategories_API.json
Comment (ravbhatnagar) [inline specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/diagnosticsSettingsCategories_API.json]: I don't think there should be location or tags here. Isnt this a proxy resource.

**Review by ravbhatnagar — CHANGES_REQUESTED** [review]
@vcanaa - please take a look

File: specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/diagnosticsSettingsCategories_API.json
Comment (vcanaa) [reply specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/diagnosticsSettingsCategories_API.json]: It is a proxy resource, but these are populated anyway when the request comes back. Also the resource class is shared by everything. It has been that way for a long time.

I believe we could have another class for proxy resource, but I prefer addressing that in a different release.

File: specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/diagnosticsSettingsCategories_API.json
Comment (vcanaa) [reply specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/diagnosticsSettingsCategories_API.json]: This is explicit in the example with a logic app.

File: specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/diagnosticsSettingsCategories_API.json
Comment (vcanaa) [reply specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/diagnosticsSettingsCategories_API.json]: I consider Enum bad practice when serialization is involved. If we introduce a new type, customers will be forced to update the sdk or the serialization will break.

Comment (ravbhatnagar) [reply specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/diagnosticsSettingsCategories_API.json]: Yes and that is the recommendation. Enum will make your API more usable. Else how will someone know when you support new types? When you support new types, you add the enum value and increment the APi version

File: specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/diagnosticsSettingsCategories_API.json
Comment (ravbhatnagar) [reply specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/diagnosticsSettingsCategories_API.json]: Can we get an ACK in email with a workitem that this will be addressed in the next release? Apologies, but this is the only way to get the APIs cleaned up across Azure plat. I wont remember this comment in a few months and no one else on your team will :). So it will most likely be missed in the next rev also.

File: specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/diagnosticsSettingsCategories_API.json
Comment (ravbhatnagar) [reply specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/diagnosticsSettingsCategories_API.json]: Is it too difficult to add it to the description? SDK users rely on description, API docs rely on description.

Comment (vcanaa) [reply specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/diagnosticsSettingsCategories_API.json]: Any change in the description breaks the generation, so it requires more work that touches all our classes. I created a task on us to fix that.

File: specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/diagnosticsSettingsCategories_API.json
Comment (vcanaa) [reply specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/diagnosticsSettingsCategories_API.json]: Created a task for this

File: specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/diagnosticsSettingsCategories_API.json
Comment (vcanaa) [reply specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/diagnosticsSettingsCategories_API.json]: I don't like enums in serialization, but I am taking this feedback since more people seem to like them.

**Comment by ravbhatnagar** [comment]
Discussed over email and there are workitems created to address the feedback. Approving for now.

**Review by ravbhatnagar — APPROVED** [review]

File: specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/examples/getDiagnosticSetting.json
Comment (sergey-shandar) [inline specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/examples/getDiagnosticSetting.json]: The response model `DiagnosticSettings` doesn't have `ServiceBusRuleId` property and no additional properties are accepted.

File: specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/examples/createOrUpdateDiagnosticSetting.json
Comment (sergey-shandar) [inline specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/examples/createOrUpdateDiagnosticSetting.json]: parameter `name` is missing.

File: specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/diagnosticsSettingsCategories_API.json
Comment (sergey-shandar) [inline specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/diagnosticsSettingsCategories_API.json]: Operations API should be implemented.

File: specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/diagnosticsSettings_API.json
Comment (sergey-shandar) [inline specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/diagnosticsSettings_API.json]: Operations API should be implemented.

File: specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/diagnosticsSettings_API.json
Comment (sergey-shandar) [inline specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/diagnosticsSettings_API.json]: The name of the model `DiagnosticSettings` equals to an operation group name. For example, `DiagnosticSettings_Get`. It means, an SDK type for the model will resolved as `DiagnosticSettingsModel`. Please, consider changing the model name.

**Comment by sergey-shandar** [comment]
@ravbhatnagar a new commit has a new file `specification/monitor/resource-manager/microsoft.insights/2015-04-01/operations_API.json`. It looks like we need your approval again.

**Review by sergey-shandar — COMMENTED** [review]
Preliminary comments. I haven't looked the new operation api file yet, I'm waiting for ARM review.

**Comment by ravbhatnagar** [comment]
Looks good. @sergey-shandar @veronicagg - FYI that the operations API and the response could be converted to a linter rule since its standard.

**Comment by sergey-shandar** [comment]
@vcanaa let me know when it's ready for review. It's hard to review a moving target :-)

**Comment by vcanaa** [comment]
This is ready for review.

**Comment by sergey-shandar** [comment]
@vcanaa thank you :-)

**Comment by vcanaa** [comment]
I was not sure if you were waiting for the sdk PR or not until a saw a new tag ReadyForSdkReview. So I sent a new PR for the sdk now.

**Comment by vcanaa** [comment]
I renamed the spec to:
[Monitor] Adding Operations API and making DiagnosticSettings ProxyOnly resource

So it is more informative.

File: specification/monitor/resource-manager/microsoft.insights/2015-04-01/examples/OperationList.json
Comment (sergey-shandar) [inline specification/monitor/resource-manager/microsoft.insights/2015-04-01/examples/OperationList.json]: The "properties" property is not allowed according to "Operation" definition.

File: specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/examples/getDiagnosticSettingsCategory.json
Comment (sergey-shandar) [inline specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/examples/getDiagnosticSettingsCategory.json]: The parameter `resourceUri` starts with a forward slash and the path template: "/{resourceUri}/providers/microsoft.insights/diagnosticSettingsCategories/{name}" contains a forward slash before the parameter starts. This will cause double forward slashes  in the request url. Thus making it incorrect. Please rectify the example.'

File: specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/examples/getDiagnosticSettingsCategory.json
Comment (sergey-shandar) [inline specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/examples/getDiagnosticSettingsCategory.json]: `tag` and `location` aren't allowed in `#/definitions/DiagnosticSettingsCategoryResource`

File: specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/examples/listDiagnosticSettings.json
Comment (sergey-shandar) [inline specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/examples/listDiagnosticSettings.json]: The parameter resourceUri starts with a forward slash and the path template: /{resourceUri}/providers/microsoft.insights/diagnosticSettingsCategories/{name} contains a forward slash before the parameter starts. This will cause double forward slashes in the request url. Thus making it incorrect. Please rectify the example.'

File: specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/examples/listDiagnosticSettings.json
Comment (sergey-shandar) [inline specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/examples/listDiagnosticSettings.json]: `tag` and `location` aren't allowed in `#/definitions/DiagnosticSettingsCategoryResource`.

File: specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/diagnosticsSettingsCategories_API.json
Comment (sergey-shandar) [inline specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/diagnosticsSettingsCategories_API.json]: `tag` and `location` aren't allowed in `#/definitions/DiagnosticSettingsCategoryResource`.

File: specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/diagnosticsSettingsCategories_API.json
Comment (sergey-shandar) [inline specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/diagnosticsSettingsCategories_API.json]: `tag` and `location` aren't allowed in `#/definitions/DiagnosticSettingsCategoryResource`.

File: specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/examples/createOrUpdateDiagnosticSetting.json
Comment (sergey-shandar) [inline specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/examples/createOrUpdateDiagnosticSetting.json]: `tag` and `location`.

File: specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/diagnosticsSettings_API.json
Comment (sergey-shandar) [inline specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/diagnosticsSettings_API.json]: `tag` and `location` shouldn't be in the response according to `DiagnosticSettingsResource`

File: specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/diagnosticsSettingsCategories_API.json
Comment (sergey-shandar) [inline specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/diagnosticsSettingsCategories_API.json]: Add description.

**Review by sergey-shandar — CHANGES_REQUESTED** [review]
Currently, the main problem is models which don't have `tag` and `location` as properties but examples do have them.

**Review by sergey-shandar — APPROVED** [review]
As an exception.

**Pull request merged**
